libraries/utils/generics/applicationlogger.cpp

```diff
@@ -0,0 +1,34 @@
+#include "applicationlogger.h"
+using namespace UTILSLIB;
+
+#define COLOR_DEBUG "\033[32;1m"
+#define COLOR_WARN "\033[33;1m"
+#define COLOR_CRITICAL "\033[31;1m"
+#define COLOR_FATAL "\033[33;1m"
+#define COLOR_RESET "\033[0m"
+
+#define LOG_WRITE(OUTPUT, COLOR, LEVEL, MSG) OUTPUT << COLOR << \
+" " LEVEL " " << MSG << COLOR_RESET << "\n"
+
+ApplicationLogger::ApplicationLogger()
+{
+
+}
+
+void ApplicationLogger::myCustomLogWriter(QtMsgType type,
```

Comment:
As @larsoner pointed out here #282 we need to consider multithreading applications as well. In order to make `myCustomLogWriter` thread safe you can for example use a QMutexLogger:
```
void ApplicationLogger::myCustomLogWriter(...) {
   static QMutex mutex;
   QMutexLocker lock(&mutex);
   
   switch (type) {
   ...
```